dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/security/impl/ldap/LdapService.java

```diff
@@ -119,7 +120,16 @@ public String ldapLogin(String userId, String userPwd) {
         } catch (NamingException e) {
             logger.error("ldap search error", e);
             return null;
+        } finally {
+            try {
+                if (ctx != null) {
+                    ctx.close();
+                }
```

Comment:
I think we should not use `if` in here since we have used `try catch`.

Reply:
The reason why we use `try catch` is `ctx.close()` will throw `NamingException`. Not using `if` is ok but IDE will remind `Method invocation 'close' may produce 'NullPointerException'`.  And not using `if` maybe log will show some `NullPointerException`, I think is not the best solution. WDYT?

Reply:
That makes sense to me.